client/src/components/Markdown/MarkdownDialog.vue

```diff
@@ -1,3 +1,223 @@
+<script setup lang="ts">
+import axios from "axios";
+import BootstrapVue from "bootstrap-vue";
+import Vue, { computed, ref } from "vue";
+
+import { getAppRoot } from "@/onload/loadConfig";
+import { getCurrentGalaxyHistory } from "@/utils/data";
+
+import MarkdownSelector from "./MarkdownSelector.vue";
+import MarkdownVisualization from "./MarkdownVisualization.vue";
+import DataDialog from "@/components/DataDialog/DataDialog.vue";
+import BasicSelectionDialog from "@/components/SelectionDialog/BasicSelectionDialog.vue";
+import DatasetCollectionDialog from "@/components/SelectionDialog/DatasetCollectionDialog.vue";
+
+Vue.use(BootstrapVue);
+
+interface MarkdownDialogProps {
+    argumentName?: string;
+    argumentType?: string;
+    argumentPayload?: object;
+    labels: string[];
+    useLabels: boolean;
+}
+
+const props = withDefaults(defineProps<MarkdownDialogProps>(), {
+    argumentName: null,
+    argumentType: null,
+    argumentPayload: null,
+});
+
+const emit = defineEmits<{
+    (e: "onCancel"): void;
+    (e: "onInsert", value: string): void;
+}>();
+
+interface SelectTitles {
+    labelTitle: string;
+}
+
+type SelectType = "job_id" | "invocation_id" | "history_dataset_id" | "history_dataset_collection_id";
+
+const selectorConfig = {
+    job_id: {
+        labelTitle: "Step",
+    },
+    invocation_id: {
+        labelTitle: "Step",
+    },
+    history_dataset_id: {
+        labelTitle: "Output",
+    },
+    history_dataset_collection_id: {
+        labelTitle: "Output",
+    },
+};
+
+const jobsUrl = `${getAppRoot()}api/jobs`;
+const workflowsUrl = `${getAppRoot()}api/workflows`;
+const invocationsUrl = `${getAppRoot()}api/invocations`;
+const historiesUrl = `${getAppRoot()}api/histories?view=detailed&q=published&qv=True`;
+
+const selectedShow = ref(false);
+const visualizationShow = ref(false);
+const workflowShow = ref(false);
+const historyShow = ref(false);
+const jobShow = ref(false);
+const invocationShow = ref(false);
+const dataShow = ref(false);
+const dataCollectionShow = ref(false);
+const dataHistoryId = ref<string | null>(null as string | null);
+
+const selectedLabelTitle = computed(() => {
+    const config: SelectTitles = selectorConfig[props.argumentType as SelectType] as SelectTitles;
+    return (config && config.labelTitle) || "Select Label";
+});
+
+function getInvocations() {
+    return axios.get(invocationsUrl);
+}
+
+function getJobs() {
+    return axios.get(jobsUrl);
+}
+
+function getWorkflows() {
+    return axios.get(workflowsUrl);
+}
+
+function getHistories() {
+    return axios.get(historiesUrl);
+}
```

Comment:
It would be nice to move these under the respective module in `client/src/api`, so they can be replaced with the proper fetcher (when ready) in one place, return proper types, and handle errors consistently. What do you think?